src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,7 +26,35 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        String[] splitFormOfAddress = trimmedAddress.split(",");
+        Block blockAddress ;
+        Street streetAddess ;
+        Unit unitAddress;
+        PostalCode postalCode;
+
+        if(splitFormOfAddress.length == 4 ){
```

Comment:
This is kinda weird. You assume the positions of the fields are fixed.
While in this case, it may be true, it is not generally advised to assume such defaults.